asserts/snapasserts/validation_sets.go

```diff
@@ -467,10 +467,16 @@ func (v *ValidationSets) constraintsForSnap(snapRef naming.SnapRef) *snapContrai
 // presence of the given snap as required and the required revision (or
 // snap.R(0) if no specific revision is required).
 // The method assumes that validation sets are not in conflict.
-func (v *ValidationSets) CheckPresenceRequired(snapRef naming.SnapRef) ([]string, snap.Revision) {
+func (v *ValidationSets) CheckPresenceRequired(snapRef naming.SnapRef) ([]string, snap.Revision, error) {
 	cstrs := v.constraintsForSnap(snapRef)
-	if cstrs == nil || cstrs.presence != asserts.PresenceRequired {
-		return nil, unspecifiedRevision
+	if cstrs == nil {
+		return nil, unspecifiedRevision, nil
+	}
+	if cstrs.presence == asserts.PresenceInvalid {
+		return nil, unspecifiedRevision, fmt.Errorf("presence of snap %q is %s", snapRef.SnapName(), cstrs.presence)
```

Comment:
maybe we want to introduce a specific error struct:  PresenceConstraintError

Reply:
Done.